qiskit/transpiler/passes/layout/vf2_utils.py

```diff
@@ -78,33 +82,46 @@ def _visit(dag, weight, wire_map):
     return im_graph, im_graph_node_map, reverse_im_graph_node_map
 
 
-def score_layout(avg_error_map, layout, bit_map, reverse_bit_map, im_graph, strict_direction=False):
+def score_layout(
+    avg_error_map,
+    layout_mapping,
+    bit_map,
+    _reverse_bit_map,
+    im_graph,
+    strict_direction=False,
+    run_in_parallel=True,
+):
     """Score a layout given an average error map."""
-    bits = layout.get_virtual_bits()
-    fidelity = 1
-    for bit, node_index in bit_map.items():
-        gate_count = sum(im_graph[node_index].values())
-        error_rate = avg_error_map.get((bits[bit],))
-        if error_rate is not None:
-            fidelity *= (1 - avg_error_map[(bits[bit],)]) ** gate_count
-    for edge in im_graph.edge_index_map().values():
-        gate_count = sum(edge[2].values())
-        qargs = (bits[reverse_bit_map[edge[0]]], bits[reverse_bit_map[edge[1]]])
-        if not strict_direction and qargs not in avg_error_map:
-            qargs = (qargs[1], qargs[0])
-        if qargs in avg_error_map:
-            fidelity *= (1 - avg_error_map[qargs]) ** gate_count
-    return 1 - fidelity
+    if layout_mapping:
+        size = max(max(layout_mapping), max(layout_mapping.values()))
+    else:
+        size = 0
+    nlayout = NLayout(layout_mapping, size + 1, size + 1)
+    bit_list = np.zeros(len(im_graph), dtype=np.int32)
+    for node_index in bit_map.values():
+        bit_list[node_index] = sum(im_graph[node_index].values())
+    edge_list = {
+        (edge[0], edge[1]): sum(edge[2].values()) for edge in im_graph.edge_index_map().values()
+    }
+    return vf2_layout.score_layout(
+        bit_list, edge_list, avg_error_map, nlayout, strict_direction, run_in_parallel
+    )
 
 
 def build_average_error_map(target, properties, coupling_map):
     """Build an average error map used for scoring layouts pre-basis translation."""
-    avg_map = {}
     num_qubits = 0
-    if coupling_map is not None:
+    if target is not None:
+        num_qubits = target.num_qubits
+        avg_map = ErrorMap(len(target.qargs))
+    elif coupling_map is not None:
         num_qubits = coupling_map.size()
+        avg_map = ErrorMap(num_qubits + coupling_map.graph.num_edges())
```

Comment:
Good catch, I think in practice any situation where neither of these if statements are hit we'd just return `None` because built will never be True. But, it's better to be through and avoid a hard error.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9026/commits/7c387239d4090500200c1d193b5b08114dda558f